shenyu-plugin/shenyu-plugin-base/src/main/java/org/apache/shenyu/plugin/base/trie/ShenyuTrie.java

```diff
@@ -135,9 +135,6 @@ public <T> void putNode(final String uriPath, final T source, final TrieCacheTyp
             selectorData = (SelectorData) source;
             node = keyRootMap.computeIfAbsent(selectorData.getPluginName(), key -> new ShenyuTrieNode("/", "/", false));
         }
-        if (TrieMatchModeEnum.PATH_PATTERN.equals(matchMode)) {
```

Comment:
this is duplicate check？

Reply:
![image](https://github.com/apache/shenyu/assets/15926669/7e2efa29-f7db-4b1e-b879-96c70352d272)
It's been checked here